### PR TITLE
[Typing][B-73] Add type annotations for `python/paddle/sparse/nn/layer/norm.py`

### PR DESCRIPTION
**Pull request by megemini**
### PR Category

User Experience

### PR Types

Improvements

### Description

类型标注：

- `python/paddle/sparse/nn/layer/norm.py`

### Related links
 
- https://github.com/PaddlePaddle/Paddle/issues/65008

@SigureMo @megemini

### CODE REVIEW

**Comment by megemini** [comment]
### Update 20240709

特别说明：

`python/paddle/sparse/nn/layer/norm.py` 中的 `class BatchNorm(paddle.nn.BatchNorm1D):` 可能实现逻辑有问题 ～

`BatchNorm` 的初始化参数为：
- `data_format: Literal["NDHWC", "NHWC"] = "NDHWC",` 

但 `BatchNorm1D` 的参数为：
- `data_format: DataLayout1D = 'NCL',`

**Review by SigureMo — APPROVED** [review]
LGTMeow
